jsonschema-core.md

```diff
@@ -2071,71 +2071,44 @@ Omitting this keyword has the same assertion behavior as an empty schema.
+information MUST use this terminology to describe it. Conversely, output
+specifications which use these terms MUST maintain their meaning.
 
-### Output Formats
+### Evaluation path
 
-This specification defines three output formats. See the "Output Structure"
-section for the requirements of each format.
-
-- *Flag*: A boolean which simply indicates the overall validation result with no
-  further details.
-- *List*: Provides validation information in a flat list structure.
-
-Hierarchical: Provides validation information in a hierarchical structure that
-follows the evaluation paths generated while processing the schema.
-
-An implementation MUST provide the "flag" format and SHOULD provide at least one
-of the "list" or "hierarchical" formats. Implementations SHOULD specify in their
-documentation which formats they support.
-
-### Minimum Information
-
-Beyond the simplistic "flag" output, additional information is useful to aid in
-debugging a schema or instance. Each sub-result SHOULD contain the information
-contained within this section at a minimum.
-
-A single object that contains all of these components is considered an output
-unit.
-
-Implementations MAY elect to provide additional information.
-
-#### Evaluation path
-
-The evaluation path to the schema object that produced the output unit. The
-value MUST be expressed as a JSON Pointer, and it MUST include any by-reference
-applicators such as `$ref` or `$dynamicRef`.[^13]
-
-[^13]: The schema may not actually have a value at the location indicated by
-this pointer. It is provided as an indication of the traversal path only.
+The evaluation path is the set of keys, starting from the schema root, through
+which evaluation passes to reach the schema object that produced a specific
+result. The value MUST be expressed as a JSON Pointer, and it MUST include any
+by-reference applicators such as `$ref` or `$dynamicRef`.
 
 ```
 /properties/width/$ref/allOf/1
 ```
 
-Note that this pointer may not be resolvable by the normal JSON Pointer process
-due to the inclusion of these by-reference applicator keywords.
+Note that this pointer may not be resolvable on the root schema by the normal
+JSON Pointer process. It is intended as an indication of the traversal path
+only.
 
-The JSON key for this information is "evaluationPath".
+When represented in JSON, the key for this information MUST be "evaluationPath".
 
-#### Schema Location
+### Schema Location
 
-The absolute, dereferenced location of the schema object that produced the
-output unit. The value MUST be expressed using the canonical IRI of the relevant
-schema resource plus a JSON Pointer fragment that indicates the schema object
-that produced the output. It MUST NOT include by-reference applicators such as
-`$ref` or `$dynamicRef`.[^14]
+The schema location is the absolute, dereferenced location of the schema object
+that produced a result. The value MUST be expressed using the canonical IRI of
```

Comment:
I can imagine a realistic exception to this "MUST". If the schema is used as a file, it's canonical IRI would include a full filesystem path. An implementation that doesn't want to expose full file paths for security reasons might choose to express the schema location as a relative-reference IRI instead of a full non-relative IRI.

Reply:
The point of this being a fully dereferenced IRI is so that the consumer can identify the exact (sub)schema immediately without having to perform any dereferencing operations.  That can't be done with a relative IRI.

However, given that the spec is now only defining the term and not requiring the value's inclusion, I think it's reasonable that an output spec could allow returning relative IRIs, but it would need to call it something else (e.g. `relativeSchemaLocation`).

Reply:
I think I prefer to keep the MUST here and possibly add a note to the security considerations.  (We should utilize that section more, actually.)